### PR TITLE
API doc improvement Dropout and SoftmaxActivation

### PR DESCRIPTION
**Pull request by jiajiechen**
Add Dropout and SoftmaxActivation.
@zackchase @madjam @nswamy @Roshrini @mli 

Thanks for reviewing my PR!

Note: Confirmed with Eric SoftmaxActivation will be deprecated.

### CODE REVIEW

File: src/operator/dropout.cc [hunk 1]
```diff
@@ -31,15 +31,35 @@ Operator *DropoutProp::CreateOperatorEx(Context ctx, std::vector<TShape> *in_sha
...
+.describe(R"(Applies dropout operation to input array.
+
+- During training, each element of the input is randomly set to zero with probability p.
+  The whole array is rescaled by :math:`1/(1-p)` to keep the expectated sum of the input unchanged.
+
+- During testing, this operator does not change the input.
+
+Example::
+
+  >>> mx.random.seed(998)
```
Comment (nswamy) [inline src/operator/dropout.cc]: Python examples should go the ndarray_doc.py if this is generic and need not be python specific. remove all the language specific constructs.

File: src/operator/softmax_activation.cc [hunk 2]
```diff
@@ -22,15 +22,32 @@ Operator *SoftmaxActivationProp::CreateOperator(Context ctx) const {
...
+
+If `mode` = ``instance``, this operator will compute a softmax for each instance in the batch.
+This is the default mode.
+
+If `mode` = ``channel``, this operator will compute a k-class softmax at each position
+of each instance, where `k` = ``num_channel``. This mode can only be used when the input array
+has at least 3 dimensions.
+This can be used for fully convolutional network, image segmentation, etc.
+
+Example::
```
Comment (nswamy) [inline src/operator/softmax_activation.cc]: same as above, python examples shouldn't be here

File: src/operator/softmax_activation-inl.h [hunk 3]
```diff
@@ -36,11 +36,9 @@ struct SoftmaxActivationParam : public dmlc::Parameter<SoftmaxActivationParam> {
     .add_enum("instance", softmax_activation::kInstance)
     .add_enum("channel", softmax_activation::kChannel)
     .set_default(softmax_activation::kInstance)
-    .describe("Softmax Mode. If set to instance, this operator will compute a "
-    "softmax for each instance in the batch; this is the default mode. "
-    "If set to channel, this operator will compute a num_channel-class softmax at "
-    "each position of each instance; this can be used for fully convolutional network, "
-    "image segmentation, etc.");
+    .describe("Specifies how to compute the softmax. If set to ``instance``, "
```
Comment (nswamy) [inline src/operator/softmax_activation-inl.h]: add a note that this API is deprecated.

File: src/operator/dropout.cc [hunk 4]
```diff
@@ -31,15 +31,35 @@ Operator *DropoutProp::CreateOperatorEx(Context ctx, std::vector<TShape> *in_sha
...
 MXNET_REGISTER_OP_PROPERTY(Dropout, DropoutProp)
-.describe(R"(Applies dropout to input.
-During training, each element of the input is randomly set to zero with probability p.
-And then the whole tensor is rescaled by 1/(1-p) to keep the expectation the same as
-before applying dropout. During the test time, this behaves as an identity map.
-)")
-.add_argument("data", "NDArray-or-Symbol", "Input data to dropout.")
+.describe(R"(Applies dropout operation to input array.
+
+- During training, each element of the input is randomly set to zero with probability p.
```
Comment (madjam) [inline src/operator/dropout.cc]: I'd omit "randomly" since you have probability with p in there.

File: src/operator/softmax_activation-inl.h (hunk 3 above)
Comment (jiajiechen) [inline src/operator/softmax_activation-inl.h]: The note is added in softmax_activation.cc.

File: src/operator/dropout.cc (hunk 1 above)
Comment (jiajiechen) [inline src/operator/dropout.cc]: Ok!

File: src/operator/dropout.cc [hunk 5]
```diff
@@ -31,15 +31,35 @@ Operator *DropoutProp::CreateOperatorEx(Context ctx, std::vector<TShape> *in_sha
...
+  The whole array is rescaled by :math:`1/(1-p)` to keep the expectated sum of the input unchanged.
```
Comment (zackchase) [inline src/operator/dropout.cc]: "expectated" -> "expected"  (expectated not a word in English although it apparently was a word in Latin :P)

File: src/operator/dropout.cc [hunk 6]
```diff
@@ -31,15 +31,35 @@ Operator *DropoutProp::CreateOperatorEx(Context ctx, std::vector<TShape> *in_sha
...
+  [[ 3.75   0.625 -0.     2.5    8.75 ]
+   [ 2.5   -0.5    8.75   3.75   0.   ]]
+
+  # If testing
+  >>> executor.forward(is_train=False, a = input_array)
+  >>> print executor.outputs[0].asnumpy()
+  [[ 3.     0.5   -0.5    2.     7.   ]
+   [ 2.    -0.4    7.     3.     0.2  ]]
+)" ADD_FILELINE)
+.add_argument("data", "NDArray-or-Symbol", "Input array to apply dropout operation.")
```
Comment (zackchase) [inline src/operator/dropout.cc]: -> "Input array to which dropout will be applied."

File: src/operator/softmax_activation.cc [hunk 7]
```diff
@@ -22,15 +22,32 @@ Operator *SoftmaxActivationProp::CreateOperator(Context ctx) const {
...
+
+  This operator has been depreated, please use `softmax`.
+
+If `mode` = ``instance``, this operator will compute a softmax for each instance in the batch.
+This is the default mode.
+
+If `mode` = ``channel``, this operator will compute a k-class softmax at each position
+of each instance, where `k` = ``num_channel``. This mode can only be used when the input array
+has at least 3 dimensions.
+This can be used for fully convolutional network, image segmentation, etc.
```
Comment (zackchase) [inline src/operator/softmax_activation.cc]: change -> "This can be used for *a* *full-convolutional* network..."
